tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -329,20 +355,57 @@ private void AssertAuthenticationFails(MongoClientSettings settings)
         {
             return CreateDatabaseUser(client, "admin", userName, password, "readWriteAnyDatabase", mechanisms);
         }
-        
+
         private BsonDocument CreateDatabaseUser(
             MongoClient client,
             string source,
-            string userName, 
-            string password, 
-            string role, 
+            string userName,
+            string password,
+            string role,
             params string[] mechanisms)
         {
             var createUserCommand =
-                $"{{createUser: '{userName}', pwd: '{password}'," 
+                $"{{createUser: '{userName}', pwd: '{password}',"
                 + $"roles: ['{role}'], mechanisms: {mechanisms.ToJson()}}}";
             return client.GetDatabase(source).RunCommand<BsonDocument>(createUserCommand);
         }
-        
+
+        private BsonDocument CreateX509DatabaseUser(MongoClient client, string userName)
+        {
+            var createUserCommand =
+                $"{{ createUser: '{userName}'," +
+                @"   roles: [ { role: 'readWrite', db: 'test' },
+                              { role: 'userAdminAnyDatabase', db: 'admin' } 
+                    ]
+                  }";
+            return client.GetDatabase("$external").RunCommand<BsonDocument>(createUserCommand);
+        }
+
+        private List<object> DropDatabaseUser(MongoClient client, string userName, IEnumerable<string> databases)
+        {
+            var dropUserCommand = $"{{ dropUser: '{userName}'}}";
+            
+            return databases.Select(TryDropUser).ToList();
+
+            object TryDropUser(string database)
+            {
+                object result;
+                try
+                {
+                    result = client.GetDatabase(database).RunCommand<BsonDocument>(dropUserCommand);
+                }
+                catch (MongoCommandException ex) // command throws if user doesn't exist
+                {
+                    result = ex;
+                }
+                return result;
+            }
+        }
+
+        private string GetRfc2253FormattedUsernameFromX509ClientCertificate(X509Certificate2 certificate)
+        {
+            var distinguishedName = certificate.SubjectName.Name;
+            return string.Join(",",  distinguishedName.Split(',').Select(s => s.Trim())).Replace("S=", "ST=");
```

Comment:
There's a ticket about the difference between `S` and `ST`  in the dotnet repo as well: https://github.com/dotnet/runtime/issues/26443

Reply:
This line is too complex. Surely it can be refactored into something more comprehensible.

Reply:
As far as I can figure out this line could just be:

    return distinguishedName.Replace("S=", "ST=");

Reply:
Refactored for clarity: what do you think of this version?